### PR TITLE
Change `lab` schema proposal

### PR DESCRIPTION
**Pull request by williamputraintan**
- Adding `File` object for all files stored
- `ArtifactBase` will have a `name` properties that is shared between files. (Thinking might be useful, if a fastq file come to know which fastqPairArtifact to add on.)
- `RunArtifactFastqPair` file will have files that could contain multiple fastq `File` object

_I did not push the migration, so might need to migrate to test locally. If this is OK, could update the insert-test-data-10g file_

### CODE REVIEW

File: application/backend/dbschema/lab.esdl [hunk 1]
```diff
@@ -16,18 +20,20 @@ module lab {
     }
 
     type RunArtifactFastqPair extending RunArtifactBase {
-        property urlR2 -> str;
+        multi link files -> File;
```
[the remark this answers is not in the document]
Comment (andrewpatto) [reply application/backend/dbschema/lab.esdl:23]: If we want a more generic 'list of raw sequence reads' I'd call the object FastqSequences or something.. but we'd need to speak to the bioinformaticians a bit about that.

**Comment by andrewpatto** [comment]
Looks good. 

I'm not 100% convinced about the use case for `name` yet - but happy for it to be there and we can explore.

File: application/backend/dbschema/lab.esdl (hunk 1 above)
[the remark this answers is not in the document]
Comment (williamputraintan) [reply application/backend/dbschema/lab.esdl:23]: So we could always say R1 would be the forward and R2 would be the reverseFile?

**Comment by williamputraintan** [comment]
> I'm not 100% convinced about the use case for `name` yet - but happy for it to be there and we can explore.

What I am thinking that if a single `.fastq` come in, and there is no `name` attribute we would need to traverse to the File object to get the URL, then we need to extract the name to know if this new file belongs to this fastq pair or not. Alternatively, we could create an identifier based on the name to know if two fastqs belongs to the same pair.

**Comment by andrewpatto** [comment]
I think we should be doing fastq pairing long before they get into this db model..

Are people really submitting single fastq files from a pair in different submissions?

**Comment by williamputraintan** [comment]
> I think we should be doing (identification of) fastq pairing long before they get into this db model..
> 
> Are people really submitting single fastq files from a pair in different submissions?

Possibly, but not entirely sure. Possible scenario is a fastq fail while the other not, and part of the remediation it will be re-submitted as a new submissions.

I am also thinking if we sync automatically data from the bucket/dydb to this db. If we use dydb-stream/s3-notification, It will come with an individual events (per file), and the pairing would then happen at this db.

**Comment by andrewpatto** [comment]
I think if a fastq pair is going into the "sharing/release" database - then it needs to go in as a complete pair..

So if someone submitting has managed to not provide a valid forward or reverse in the pair - then the pair shouldn't be inserted (I'd defer the insertion of any entry for the pair).

I understand the streaming case - but that feels like the cart driving the horse - if a stream of individual file items don't have enough semantics (i.e. for pairing and/batching and/or whatever else it might not be able to do) then I'd consider whether streams are the right abstraction - or consider what sort of staging data structures I might need to add that in.

Otherwise we are going to have to teach every other part of the system (the sharing bits) that whenever it sees a fastq pair that perhaps sometimes the data is going to be missing. I'd prefer it the rest of the system could work on solid concrete objects - and we could isolate stuff like making the pairs etc to the 'ingestion phase'.

I mean I'm happy for there to be an 'ingestion' edsl namespace with any table/objects that are supporting needs for a streaming input process?

**Pull request merged**